sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/ImmutableLongArray.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.common;
+
+import java.util.Arrays;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class ImmutableLongArray {
+  private static final ImmutableLongArray EMPTY = new ImmutableLongArray(new long[0]);
+
+  /** Returns the empty array. */
+  public static ImmutableLongArray of() {
```

Comment:
same comments as for the double version. let's keep the API super narrow

Reply:
this one is not used actually. I added it mainly to keep symmetry with `ImmutableDoubleArray`. I could get rid of this one for sure.